### PR TITLE
Dev/ben 431

### PR DESCRIPTION
**Pull request by bLopata**
https://app.screencast.com/vgr6gPu1Skp9N

uses localstorage.

### CODE REVIEW

File: www/app/providers.tsx [hunk 1]
```diff
@@ -14,7 +15,7 @@ if (
 ) {
   posthog.init(posthogKey, {
     api_host: posthogHost,
-    capture_pageview: false, // Disable automatic pageview capture, as we capture manually
```
Comment (VVoruganti) [inline www/app/providers.tsx:17]: I think we can keep the comment

File: www/utils/swrCache.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { Cache } from 'swr';
+
+const memoryCache = new Map<string, any>();
```
Comment (VVoruganti) [inline www/utils/swrCache.ts:3]: Looks like this is just an in memory cache not actually using indexedDB

File: www/utils/swrCache.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export const cacheProvider = (cache: Readonly<Cache<any>>) => ({
+  get: (key: string) => memoryCache.get(key),
+  set: (key: string, value: any) => memoryCache.set(key, value),
+  delete: (key: string) => memoryCache.delete(key),
+  keys: () => Array.from(memoryCache.keys())[Symbol.iterator]()
+});
+
+export const clearSWRCache = () => memoryCache.clear();
```
Comment (VVoruganti) [inline www/utils/swrCache.ts:12]: based on https://swr.vercel.app/docs/advanced/cache

This might be a dangerous way to clear the cache

**Review by VVoruganti — CHANGES_REQUESTED** [review]
This doesn't seem to actually used IndexedDB and is invalidating the cache in a dangerous way according to the SWR docs. 

Should still invalidate on logout

Like how you simplified the conversationFetcher since that minimum 1 conversation is managed in the `Sidebar.tsx` already

File: www/utils/swrCache.ts (hunk 2 above)
Comment (bLopata) [reply www/utils/swrCache.ts]: Reverted to localStorage

File: www/utils/swrCache.ts (hunk 3 above)
Comment (bLopata) [reply www/utils/swrCache.ts]: Good catch, used the `mutate` strategy from the docs.

File: www/components/sidebar.tsx [hunk 4]
```diff
@@ -262,6 +263,7 @@ export default function Sidebar({
                   className="block px-4 py-2 text-sm text-gray-700 dark:text-gray-200 hover:bg-gray-100 dark:hover:bg-gray-700 w-full text-left"
                   onClick={async () => {
                     await supabase.auth.signOut();
+                    clearSWRCache();
```
Comment (VVoruganti) [inline www/components/sidebar.tsx:266]: When I logout my localStorage values are still there in app-cache

Comment (VVoruganti) [reply www/components/sidebar.tsx:266]: image

**Review by VVoruganti — APPROVED** [review]
The invalidation logic doesn't seem to be working. Otherwise looks good

**Review by VVoruganti — CHANGES_REQUESTED** [review]
Once invalidation logic works then we're good

**Comment by bLopata** [comment]
> The invalidation logic doesn't seem to be working. Otherwise looks good

Working now.

**Review by VVoruganti — APPROVED** [review]

**Pull request merged**